examples/sphere/shallow_water.jl

```diff
@@ -46,18 +45,57 @@ end
 #   body rotation or zonal flow with the corresponding geostrophic height field.
 #   This can be run with an angle α that represents the angle between the north
 #   pole and the center of the top cube panel.
+Base.@kwdef struct SteadyStateTest{FT, P} <: AbstractTest
+    params::P = Params{FT}()
+    u0::FT = 2 * pi * params.R / (12 * 86400)
+    h0::FT = 2.94e4 / params.g
+    α::FT
+end
+SteadyStateTest(α::FT) where {FT} = SteadyStateTest{FT, Params{FT}}(; α = α)
 # - A second one, called "steady_state_compact", reproduces Test Case 3 in the same
 #   reference paper. This test case gives the steady-state solution to the
 #   non-linear shallow water equations with nonlinear zonal geostrophic flow
 #   with compact support.
+Base.@kwdef struct SteadyStateCompactTest{FT, P} <: AbstractTest
+    params::P = Params{FT}()
+    u0::FT = 2 * pi * params.R / (12 * 86400)
+    h0::FT = 2.94e4 / params.g
+    ϕᵦ::FT = -30.0
+    ϕₑ::FT = 90.0
+    xₑ::FT = 0.3
```

Comment:
Inline commented documentation should suffice for this ; similar to case 4 below.